src/test/java/org/opensearch/neuralsearch/highlight/SemanticHighlighterIT.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.highlight;
+
+import static org.opensearch.neuralsearch.util.TestUtils.TEST_DIMENSION;
+import static org.opensearch.neuralsearch.util.TestUtils.TEST_SPACE_TYPE;
+import static org.opensearch.neuralsearch.util.TestUtils.createRandomVector;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Locale;
+
+import org.junit.Before;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+import org.opensearch.index.query.QueryStringQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import com.google.common.primitives.Floats;
+import lombok.SneakyThrows;
+
+public class SemanticHighlighterIT extends BaseNeuralSearchIT {
+    private static final String TEST_BASIC_INDEX_NAME = "test-semantic-highlight-index";
+    private static final String TEST_QUERY_TEXT = "artificial intelligence";
+    private static final String TEST_KNN_VECTOR_FIELD_NAME = "test-knn-vector-1";
+    private static final String TEST_TEXT_FIELD_NAME = "test-text-field";
+    private static final String TEST_CONTENT = "Machine learning is a field of artificial intelligence that uses statistical techniques. "
+        + "Natural language processing is a branch of artificial intelligence that helps computers understand human language. "
+        + "Deep learning is a subset of machine learning that uses neural networks with many layers.";
+    private final float[] testVector = createRandomVector(TEST_DIMENSION);
+
+    // Add fields to store model IDs
+    private String textEmbeddingModelId;
+    private String sentenceHighlightingModelId;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        updateClusterSettings();
+
+        // Set up models once for all tests
+        textEmbeddingModelId = prepareModel();
+        sentenceHighlightingModelId = prepareSentenceHighlightingModel();
```

Comment:
these two variable can be inside the test method, better do it now to avoid problems later if someone need to add more tests and they can just reuse class level variables with model ids.